infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerPhase.java

```diff
@@ -45,4 +48,32 @@ public BrokerPhase then(BrokerPhase next) {
    */
   @Beta
   public abstract List<ChePlugin> execute() throws InfrastructureException;
+
+  /**
+   * Start a new span using provided tracer. If tracer is null, returns null.
+   *
+   * <p>Meant to be used with {@link BrokerPhase#finishSpanIfExists(Span)}
+   */
+  public Span startTracingPhase(Tracer tracer, String spanName, String workspaceId) {
```

Comment:
Tracer and workspace ID are not changed during the life cycle of this class. Why not move them to Phase constructor  and not pass them on every call?

Reply:
I tried using @Traced annotations but could not get it to work (traces just don't show up). That could be a lack of understanding on my part wrt how tracing works.

I'll move the passed variables to instance vars on `BrokerPhase`, that's a good idea.

Reply:
If object is not injected by Guice  annotation should not work

Reply:
@garagatyi Right, of course.